plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/ProcessNotificationsBase.kt

```diff
@@ -3,27 +3,59 @@
 
 package software.aws.toolkits.jetbrains.core.notifications
 
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
 import com.intellij.notification.NotificationType
 import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.application.PathManager
 import com.intellij.openapi.components.Service
 import com.intellij.openapi.components.service
 import com.intellij.openapi.project.Project
+import software.aws.toolkits.core.utils.inputStream
 import software.aws.toolkits.jetbrains.utils.notifyStickyWithData
+import java.nio.file.Paths
+
+object NotificationMapperUtil {
+    val mapper = jacksonObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+}
 
 @Service(Service.Level.PROJECT)
-class ProcessNotificationsBase {
+class ProcessNotificationsBase(
+    private val project: Project,
+) {
     private val notifListener = mutableListOf<NotifListener>()
     init {
-        // TODO: install a listener for the polling class
+        NotificationPollingService.getInstance().addObserver { isStartup ->
+            retrieveStartupAndEmergencyNotifications(isStartup)
+        }
     }
 
-    fun getNotificationsFromFile() {
-        // TODO: returns a notification list
+    private fun getNotificationsFromFile(): NotificationsList? {
+        val path = Paths.get(PathManager.getSystemPath(), NOTIFICATIONS_PATH)
+        val content = path.inputStream().bufferedReader().use { it.readText() }
+        if (content.isEmpty()) {
+            return null
+        }
+        return NotificationMapperUtil.mapper.readValue(content)
     }
 
-    fun retrieveStartupAndEmergencyNotifications() {
-        // TODO: separates notifications into startup and emergency
-        // iterates through the 2 lists and processes each notification(if it isn't dismissed)
+    fun retrieveStartupAndEmergencyNotifications(isStartup: Boolean) {
+        val notifications = getNotificationsFromFile()
+
+        notifications?.let { notificationsList ->
+            val (startupNotifications, emergencyNotifications) = notificationsList.notifications
+                ?.partition { notification ->
+                    notification.schedule.type.equals("StartUp", ignoreCase = true)
+                }
+                ?: Pair(emptyList(), emptyList())
+
+            val dismissalState = NotificationDismissalState.getInstance()
+            (if (isStartup) startupNotifications else emptyList())
```

Comment:
Should this part handle the fact that startup notifications should be shown only once per session?

Reply:
nit: can we move the handling into separate list handling for readability

Reply:
yes and yes, I was passing the boolean in from the polling service but I think the flag can live here instead.